src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStoragePublisher.java

```diff
@@ -216,12 +223,32 @@ public String getStorageCredentialId() {
     }
 
     public AzureCredentials.StorageAccountCredential getStorageCreds() {
-
-        if (storageCreds == null && storageCredentialId != null) {
-            return AzureCredentials.getStorageAccountCredential(this.storageCredentialId);
+        String credentialsId = this.storageCredentialId == null
+                ? (this.storageCreds == null ? null : this.storageCreds.getId())
+                : this.storageCredentialId;
+        try {
+            if (credentialsId != null) {
+                AzureCredentials.StorageAccountCredential credentials = AzureCredentials.getStorageAccountCredential(credentialsId);
+                if (credentials != null) {
+                    this.storageCreds = credentials;
+                    return credentials;
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
         }
-
-        return storageCreds;
+        if (this.storageCreds == null) {
```

Comment:
I don't understand what you're trying to acomplish. Should this method just call into AzureCredentials.getStorageAccountCredentials(credentialsId) ?
getStorageAccountCredentials shouldn't throw an exception and just an empty credential